### ISSUE TITLE
Nightly tests failing in StructuralMecahnics

### ISSUE THREAD

**Issue by philbucher** [KratosMultiphysics/Kratos#1660]
@KratosMultiphysics/structural-mechanics 
since last night some tests are failing, even though nothing obvious was changed

maybe #1648 is related to this?
@adityaghantasala did you run the tests before merging?

~~~
Running tests for StructuralMechanicsApplication
......................................Check failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
E......F...........s.................Check failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
ECheck failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
E.....Check failed in file /home/ubuntu/CompiledLibs/boost_1_59_0/boost/numeric/ublas/storage.hpp at line 195:
i < size_
E......................................sss
======================================================================
ERROR: test_execution (NightlyTests.SprismMembranePatchTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 45, in test_execution
    self.test.RunMainTemporalLoop()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 41, in RunMainTemporalLoop
    self.SolveTimeStep()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 54, in SolveTimeStep
    self.__SolveSolutionStep()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 225, in __SolveSolutionStep
    self.solver.Solve()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_static_solver.py", line 67, in Solve
    super(StaticMechanicalSolver, self).Solve()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py", line 253, in Solve
    mechanical_solution_strategy.Solve()
RuntimeError: Error: bad index
in kratos/solving_strategies/strategies/residualbased_linear_strategy.h:477:double ResidualBasedLinearStrategy<TSparseSpace,...>::Solve() [with TSparseSpace = UblasSpace<double, boost::numeric::ublas::compressed_matrix<...>, boost::numeric::ublas::vector<double> >; TDenseSpace = UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double> >; TLinearSolver = LinearSolver<UblasSpace<double, boost::numeric::ublas::compressed_matrix<...>, boost::numeric::ublas::vector<double> >, UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double> > >]


Stdout:
Computing model part:
-computing_domain- model part
    Number of tables : 0
    Number of sub model parts : 0

    Mesh 0 : 
        Number of Nodes      : 16
        Number of Properties : 1
        Number of Elements   : 10
        Number of Conditions : 0


{
    "model_part_name": "Parts_Parts_Auto1",
    "variable_name": "DISPLACEMENT",
    "interval": [
        0.0,
        1e30
    ],
    "value": [
        "1.0e-7 * (x + y/2)",
        "1.0e-7 * (y + x/2)",
        0.0
    ],
    "tolerance_rank": 3,
    "mesh_id": 0,
    "local_axes": {}
}

======================================================================
ERROR: test_execution (NightlyTests.SprismBendingPatchTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 45, in test_execution
    self.test.RunMainTemporalLoop()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 41, in RunMainTemporalLoop
    self.SolveTimeStep()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 54, in SolveTimeStep
    self.__SolveSolutionStep()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 225, in __SolveSolutionStep
    self.solver.Solve()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_static_solver.py", line 67, in Solve
    super(StaticMechanicalSolver, self).Solve()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py", line 253, in Solve
    mechanical_solution_strategy.Solve()
RuntimeError: Error: bad index
in kratos/solving_strategies/strategies/residualbased_linear_strategy.h:477:double ResidualBasedLinearStrategy<TSparseSpace,...>::Solve() [with TSparseSpace = UblasSpace<double, boost::numeric::ublas::compressed_matrix<...>, boost::numeric::ublas::vector<double> >; TDenseSpace = UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double> >; TLinearSolver = LinearSolver<UblasSpace<double, boost::numeric::ublas::compressed_matrix<...>, boost::numeric::ublas::vector<double> >, UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double> > >]


Stdout:
Computing model part:
-computing_domain- model part
    Number of tables : 0
    Number of sub model parts : 0

    Mesh 0 : 
        Number of Nodes      : 16
        Number of Properties : 1
        Number of Elements   : 10
        Number of Conditions : 0


{
    "model_part_name": "Parts_Parts_Auto1",
    "variable_name": "DISPLACEMENT",
    "interval": [
        0.0,
        1e30
    ],
    "value": [
        "-1.0e-7 * (z - 0.0005) * (x + y/2)",
        "-1.0e-7 * (z - 0.0005) * (y + x/2)",
        "0.5 * 1.0e-7 * (x ** 2 + x * y + y ** 2)"
    ],
    "tolerance_rank": 3,
    "mesh_id": 0,
    "local_axes": {}
}

======================================================================
ERROR: test_TL_2D_quadrilateral (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 459, in test_TL_2D_quadrilateral
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_TL_2D_triangle (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 419, in test_TL_2D_triangle
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_TL_3D_hexa (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 510, in test_TL_3D_hexa
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_UL_2D_quadrilateral (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 588, in test_UL_2D_quadrilateral
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_UL_2D_triangle (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 547, in test_UL_2D_triangle
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_UL_3D_hexa (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 640, in test_UL_3D_hexa
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_compare_TL_UL_2D_triangle (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 366, in test_compare_TL_UL_2D_triangle
    self._solve_with_strategy(tl_strategy, tl_lhs, iter)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 161, in _solve_with_strategy
    strategy.SolveSolutionStep()
IndexError: bad index

======================================================================
ERROR: test_formfinding (test_patch_test_formfinding.TestPatchTestFormfinding)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_formfinding.py", line 128, in test_formfinding
    False) # Do PostProcessing for GiD?
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_formfinding.py", line 115, in _execute_formfinding_test
    self._solve(mp)
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_formfinding.py", line 91, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
FAIL: test_execution (NightlyTests.Simple3D2NTrussTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 45, in test_execution
    self.test.RunMainTemporalLoop()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 42, in RunMainTemporalLoop
    self.FinalizeTimeStep()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 57, in FinalizeTimeStep
    self.__ExecuteFinalizeSolutionStep()
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py", line 235, in __ExecuteFinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/ubuntu/Kratos/kratos/python_scripts/from_json_check_result_process.py", line 140, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.0015960661694407463 != 0.0, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 2 REACTION_Y results.

Stdout:
Computing model part:
-computing_domain- model part
    Number of tables : 0
    Number of sub model parts : 0

    Mesh 0 : 
        Number of Nodes      : 2
        Number of Properties : 2
        Number of Elements   : 1
        Number of Conditions : 1



----------------------------------------------------------------------
Ran 130 tests in 191.154s

FAILED (failures=1, errors=10, skipped=4)
~~~

**Comment by KlausBSautter** [comment]
sry, I set the tolerance for one of my tests too low. should be fixed with my latest pr

**Comment by philbucher** [comment]
fixed in #1666
